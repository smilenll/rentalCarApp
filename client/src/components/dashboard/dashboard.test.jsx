import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Dashboard from './dashboard';
import DashboardTable from './dashboard-table';

const mockStore = configureStore([]);

let store;
let component;
const ContractReducers = {
  loading: false,
  allContracts: {},
  error: '',
};
const AmortizationReducers = {

  loading: false,
  allAmortizationFilters: {},
  error: '',
};

beforeEach(() => {

});
describe('Dashboard component', () => {
  store = mockStore({
    ContractReducers,
    AmortizationReducers,
  });

  component = renderer.create(
    <Provider store={store}>
      <Dashboard />
    </Provider>,
  );


  it('should be render correct', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render contracts from Redux store', () => {
    const findContracts = component.root.findByType(DashboardTable).props.contracts;

    expect(findContracts).toBe(ContractReducers);
  });

  it('should supply amortizations to DashboardTable', () => {
    const findAmortizations = component.root.findByType(DashboardTable).props.amortizations;

    expect(findAmortizations).toBe(AmortizationReducers);
  });
});
