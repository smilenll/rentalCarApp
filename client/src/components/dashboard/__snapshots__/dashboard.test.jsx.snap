// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should be render correct 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <table
      className="table table-hover table-dark"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Car
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="First & Last name"
          >
            Customer
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip "
            scope="col"
            title="First & Last name"
          >
            From
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated return date"
          >
            To
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated days"
          >
            days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated price per day"
          >
            price $/days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Days until today"
          >
            On day
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="The price per day which customer has to pay today"
          >
            Current price
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Total price for now"
          >
            Total price
          </th>
          <th
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Dashboard component should render Dashboard cu 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <table
      className="table table-hover table-dark"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Car
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="First & Last name"
          >
            Customer
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip "
            scope="col"
            title="First & Last name"
          >
            From
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated return date"
          >
            To
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated days"
          >
            days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated price per day"
          >
            price $/days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Days until today"
          >
            On day
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="The price per day which customer has to pay today"
          >
            Current price
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Total price for now"
          >
            Total price
          </th>
          <th
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Dashboard component should render contracts from Redux store 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <table
      className="table table-hover table-dark"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Car
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="First & Last name"
          >
            Customer
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip "
            scope="col"
            title="First & Last name"
          >
            From
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated return date"
          >
            To
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated days"
          >
            days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated price per day"
          >
            price $/days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Days until today"
          >
            On day
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="The price per day which customer has to pay today"
          >
            Current price
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Total price for now"
          >
            Total price
          </th>
          <th
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Dashboard component should render loading 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <table
      className="table table-hover table-dark"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Car
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="First & Last name"
          >
            Customer
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip "
            scope="col"
            title="First & Last name"
          >
            From
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated return date"
          >
            To
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated days"
          >
            days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Estimated price per day"
          >
            price $/days
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Days until today"
          >
            On day
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="The price per day which customer has to pay today"
          >
            Current price
          </th>
          <th
            data-placement="top"
            data-toggle="tooltip"
            scope="col"
            title="Total price for now"
          >
            Total price
          </th>
          <th
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
